api/types/kustomization.go

```diff
@@ -222,10 +224,13 @@ func (k *Kustomization) CheckDeprecatedFields() *[]string {
 // moving content of deprecated fields to newer
 // fields.
 func (k *Kustomization) FixKustomization() {
+	var warnings []string
 	if k.Kind == "" {
+		warnings = append(warnings, deprecationWarningKind)
```

Comment:
Why are you adding the warning message here instead of in the `CheckDeprecatedFields()` func?